staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -146,14 +147,25 @@ type Store struct {
 	// specific cases where storage of the value is not appropriate, since
 	// they cannot be watched.
 	Decorator ObjectFunc
+
 	// CreateStrategy implements resource-specific behavior during creation.
 	CreateStrategy rest.RESTCreateStrategy
+	// BeginCreate is an optional hook that returns a "transaction-like"
+	// commit/revert function which will be called at the end of operation,
+	// but before AfterCreate and Decorator, indicating via the argument
+	// whether the operation succeeded.  Almost nobody should use this hook.
+	BeginCreate func(ctx context.Context, obj runtime.Object, options *metav1.CreateOptions) (func(context.Context, bool), error)
```

Comment:
nit: I have a issue with being able to read functions returning functions.  Can we name this function type please?  `type MutatingCallbackFunc func(context.Context, succeeded bool)`

Reply:
nit: worth noting that the hook is not called if BeginCreate returns an error, so it cannot do all compensation logic.

Reply:
Are you attached to that name?  I went with `FinishFunc` (Begin <-> Finish), but I'll go with your pref.

Will be in next push